internal/rbd/rbd_healer.go

```diff
@@ -70,11 +74,47 @@ func getSecret(c *k8s.Clientset, ns, name string) (map[string]string, error) {
 	return deviceSecret, nil
 }
 
+// formatStagingTargetPath returns the path where the volume is expected to be
+// mounted (or the block-device is attached/mapped). Different Kubernetes
+// version use different paths.
+func formatStagingTargetPath(c *k8s.Clientset, pv *v1.PersistentVolume, stagingPath string) (string, error) {
+	// Kubernetes 1.24+ uses a hash of the volume-id in the path name
+	unique := sha256.Sum256([]byte(pv.Spec.CSI.VolumeHandle))
+	targetPath := filepath.Join(stagingPath, pv.Spec.CSI.Driver, fmt.Sprintf("%x", unique), "globalmount")
+
+	version, err := c.Discovery().ServerVersion()
+	if err != nil {
+		return "", fmt.Errorf("failed to detect Kubernetes version: %w", err)
+	}
+
+	major, err := strconv.Atoi(version.Major)
+	if err != nil {
+		return "", fmt.Errorf("failed to convert Kubernetes major version %q to int: %w", version.Major, err)
+	}
+
+	minor, err := strconv.Atoi(version.Minor)
+	if err != nil {
+		return "", fmt.Errorf("failed to convert Kubernetes minor version %q to int: %w", version.Minor, err)
+	}
```

Comment:
we already have a utility get this detail?

Reply:
Is it k8sVersionGreaterEquals() from e2e ?

Reply:
https://github.com/ceph/ceph-csi/blob/af0cd0fd80db35bb3011ca7455a9f5f49e4b2e87/internal/util/k8s/version.go#L28-L45

Reply:
Cool, addressed now.